File: tests/rust/tests-rust-testing-selenium/tests/e2e/home_tests.rs
use allure_cargotest::allure_test;
use tests_rust_testing_selenium as tests;

#[allure_test(name = "Page load fetches health and items from API")]
#[tokio::test]
async fn page_load_fetches_items() {
    tests::layer_e2e("Home", "Home", "Home load", "normal");
    allure_rust_commons::tag("smoke");
    tests::with_browser(|| async {
        let service = tests::load_config().api_health_service;
        tests::HomePage::default()
            .open_page()
            .await
            .should_show_health_text(&format!("service: {service}"))
            .await
            .should_show_item_text("Alpha")
            .await;
    })
    .await;
}

File: tests/csharp/tests-csharp-nunit-restsharp-selenium/Tests/E2e/HomeTests.cs
using Allure.Net.Commons;
using Allure.NUnit.Attributes;
using Tests;

namespace Tests.E2e;

[AllureLabel("layer", "e2e")]
[AllureEpic("Home")]
[AllureFeature("Home load")]
[AllureSeverity(SeverityLevel.normal)]
[AllureSuite("Home")]
public sealed class HomeTests : TestBase
{
    [Test]
    [Category("e2e")]
    [Category("smoke")]
    [AllureName("Page load fetches health and items from API")]
    public void PageLoadFetchesItems()
    {
        HomePage.OpenPage()
            .ShouldShowHealthText("service: " + Config.ApiHealthService)
            .ShouldShowItemText("Alpha");
    }
}
